src/codeu/chat/client/ClientConversation.java

```
@@ -86,10 +86,10 @@ public void showCurrent() {
     printConversation(currentSummary, userContext);
   }
 
-  public void startConversation(String title, Uuid owner) {
+  public void startConversation(String title, Uuid owner, String passHash, String salt) {
     final boolean validInputs = isValidTitle(title);
```

Checking validity of arguments is a very good practice! I'd recommend checking all of them -- at least that they are non-null, maybe more.

Normally you would want to throw an unchecked exception such as IllegalArgumentException, then let the caller (or the caller's caller, etc.) deal with how to display it.

Also, independently from that, it's a good idea to handle unexpected/error conditions *first* -- do an early return or throw, etc., so the rest of the method doesn't have to worry about it. That also makes code less nested.

---

Looked around and changed what I could/found _\*thumbs-up\*_